go/jsonclient/client.go

```diff
@@ -0,0 +1,191 @@
+package jsonclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	ct "github.com/google/certificate-transparency/go"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// JSONClient provides common functionality for interacting with a JSON server
+// that uses cryptographic signatures.
+type JSONClient struct {
+	uri        string                // the base URI of the server. e.g. http://ct.googleapis/pilot
+	httpClient *http.Client          // used to interact with the server via HTTP
+	Verifier   *ct.SignatureVerifier // nil if no public key for server available
```

Comment:
The outside world needs to interact with `Verifier` (to, er, verify) but shouldn't need to change the base URI or the client after start-of-day.  But I could add a forwarding `JSONClient.VerifySignature([]byte, tls.DigitallySigned)` if you prefer?